state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -838,30 +850,25 @@ pub mod pallet {
 			(ccm_output_principal, ccm_output_gas): (AssetAmount, AssetAmount),
 		) {
 			// Schedule the given ccm to be egressed and deposit a event.
-			let egress_id = T::EgressHandler::schedule_egress(
+			if let Ok((egress_id, egress_amount, egress_fee)) = T::EgressHandler::schedule_egress(
 				ccm_swap.destination_asset,
 				ccm_output_principal,
 				ccm_swap.destination_address.clone(),
 				Some((ccm_swap.deposit_metadata, ccm_output_gas)),
-			);
-
-			if let Some(swap_id) = ccm_swap.principal_swap_id {
-				Self::deposit_event(Event::<T>::SwapEgressScheduled {
-					swap_id,
-					egress_id,
-					asset: ccm_swap.destination_asset,
-					amount: ccm_output_principal,
-				});
-			}
-			if let Some(swap_id) = ccm_swap.gas_swap_id {
-				Self::deposit_event(Event::<T>::SwapEgressScheduled {
```

Comment:
I removed this because there's no actual egress scheduled for the gas, so it was wrong. Maybe we can extend the ccm event somehow, but that's out of scope here.